src/snowcli/cli/nativeapp/version/version_processor.py

```diff
@@ -0,0 +1,311 @@
+
+        with self.use_role(self.package_role):
+            # Now that the application package exists, create shared data
+            self._apply_package_scripts()
+
+            # Upload files from deploy root local folder to the above stage
+            self.sync_deploy_root_with_stage(self.package_role)
+
+        # Check if the version exists in a release directive(s)
+        warn_user_about_existing_rd(
+            existing_rds=self.get_existing_release_directive_info_for_version(version),
+            version=version,
+            package_name=self.package_name,
+            user_confirmation_policy=user_confirmation_policy,
+        )
+
+        # Add a new version to the app package
+        if not self.get_existing_version_info(version):
+            self.add_new_version(version=version)
+            raise typer.Exit()  # A new version created automatically has patch 0, we do not need to further increment the patch.
+
+        # Add a new patch to an existing (old) version
+        self.add_new_patch_to_version(version=version, patch=patch)
+
+
+class NativeAppVersionDropProcessor(NativeAppManager, NativeAppCommandProcessor):
+    def __init__(self, project_definition: Dict, project_root: Path):
+        super().__init__(project_definition, project_root)
+
+    def process(self, version: Optional[str], force: bool = False, *args, **kwargs):
+        """
+        Drops a version associated with an application package. If --force is provided, then no user prompts will be executed.
+        """
+        user_confirmation_policy = UserConfirmationPolicy(
+            is_force=is_force_lambda(force),
+            is_interactive_mode=is_interactive_mode_lambda(),
+            confirm_with_user=confirm_with_user_lambda,
+        )
+
+        # 1. Check for existing an existing application package
+        show_obj_row = self.get_existing_app_pkg_info()
+        if show_obj_row is None:
+            raise ApplicationPackageDoesNotExistError(self.package_name)
+        else:
+            # Check for the right owner role
+            ensure_correct_owner(
+                row=show_obj_row, role=self.package_role, obj_name=self.package_name
+            )
+
+        # 2. If the user did not pass in a version string, determine from manifest.yml
+        if version is None:
+            print(
+                dedent(
+                    f"""\
+                    Version was not provided through the CLI. Checking version in the manifest.yml instead.
+                    This step will bundle your app artifacts to determine the location of the manifest.yml file.
+                  """
+                )
+            )
+            self.build_bundle()
+            version, _ = find_version_info_in_manifest_file(self.deploy_root)
+
+        print(
+            dedent(
+                f"""\
+                    About to drop version {version} of application package {self.package_name}. Once dropped, this operation cannot be undone.
```

Comment:
Suggest you move the "Once dropped, ..." bit to ask user confirmation below. It's only relevant to the user if they can interactively stop it.